src/main/java/org/openmrs/uitestframework/page/Page.java

```diff
@@ -152,8 +152,6 @@ public void go() {
     }
 
     public WebElement findElement(By by) {
-        waiter.until(ExpectedConditions.visibilityOfElementLocated(by));
```

Comment:
> increasing the time limit of the page at runtime might get things ahead though still debugging into this

Just as I said above, the **waiter** instance takes _searching the element instance_ into an infinite loop **But** when the element is already retrieved. So even when _runtime_ lag is increased to whatever value it may be set to, the loop will still override it and the same exception will be thrown after passing the set time. Currently the set time is 120 seconds with 500 milliseconds interval

Reply:
i think we need to check whether an element is visible using boolean like isDisplayed or isEnabled commands to check whether the element is visible then ,if an element is invisible we ignore exceptions and continue

Reply:
> what causes that infinite looping?

Am investigating the actual cause for this

Reply:
> i would still maintain this method as it is and only change where it's causing such an unexpected effect as i investigate the true cause of such an effect

Am susceptible about the behaviour of the waiter instance invoked within findElement() function. Actually the element is found when the waiter instance is still waiting for the visibility of the element. Ideally If this instance is to remains being invoked then wrapping try catch block logic  would be a better way to handle the instance behaviour. cc: @kaweesi @sherrif10

Reply:
> can you try ExpectedConditions.presenceOfElementLocated(locator)?

Am looking at the behaviour of this class.

Reply:
**Removing the waiter instance resolves the bug**. However, the **pain** comes to the programmers' side working on any of the instance related to invoking  findElement() function; that when an invoked element can not be visible on a page, the **error** will miss pointing out the actual element attribute (some think like **waiting for visibility of element located by By.attributeName()**) **However**, the exception will indicate the line number where things are breaking and this tasks the contributor to figure out the element missing on the page. cc: @kaweesi @sherrif10